torch/nn/functional.py

```diff
@@ -2486,6 +2499,17 @@ def _check_size_scale_factor(dim):
                 and len(scale_factor) != dim:
             raise ValueError('scale_factor shape must match input shape. '
                              'Input is {}D, scale_factor size is {}'.format(dim, len(scale_factor)))
+        if scale_factor is not None:
+            if use_scale_factor is None:
+                is_float_scale_factor = any(not float(scale).is_integer() for scale in _ntuple(dim)(scale_factor))
+                if is_float_scale_factor:
+                    warnings.warn("The default behavior for interpolate/upsample with float scale_factor will change "
```

Comment:
Is there a particular reason we're not exposing `use_scale_factor` for upsample? (maybe bc they are deprecated, just curious).

Reply:
yes. Even align_corners is not exposed to upsample.